scripts/tests/test_wave_smoke_lifecycle.py

```
@@ -0,0 +1,289 @@
+    if shutil.which("ss"):
+        completed = subprocess.run(
+            ["ss", "-tlnp", f"sport = :{port}"],
+            text=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            timeout=5,
+            check=False,
+        )
+        return sorted({int(pid) for pid in re.findall(r"pid=(\d+)", completed.stdout)})
+    if shutil.which("fuser"):
+        completed = subprocess.run(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)